contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,226 @@
+Consider a simple web application that displays the name of a person as shown below.
+
+<iframe src="https://stackblitz.com/edit/angular-3qy4j2?embed=1&file=src/app/app.component.ts&view=preview" style="width:85%; height:380px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe> 
+
+<box type="info">
+    The sandbox above is editable. You can enter your name in the text box and see how the UI is automatically updated. You can also view and edit the code by clicking on the Both button. 
+</box>
+
+The UI of an Angular app is defined in _templates_ which contain a mix of HTML code and HTML-like custom tags. The template for the above example (defined in `person.component.html`) is given below.
+
+``` html
+<h1>About Me!</h1>
+<h2>Hello World! My name is {person.name}</h2>
+<div>
+<label>NAME: <input [(ngModel)]="person.name"/></label>
+</div>
+```
+  
+The templates used in Angular are _declarative_ because they describe _what_ the view should look like rather than giving instructions on _how_ to construct the view. In the above example, we declare what we want displayed (i.e., name) and Angular renders the view. A noteworthy feature about the UI rendered by Angular is its _dynamic behavior_.
+You will notice that as you change the name in the text box, the name rendered on the app is also updated. This is because the UI and the application data are automatically synchronized by Angular as explained [here](https://angular.io/guide/template-syntax#binding-syntax-an-overview).
+
+Angular organizes the behavior instructions (written in the TypeScript language) into what it calls _Components_. Given below is the component file `person.component.ts` of the `PersonComponent` :  
+
+```typescript
+@Component({
+  selector: 'app-person',
+  templateUrl: './person.component.html',
+  styleUrls: ['./person.component.css']
+})
+export class PersonComponent {
+  person: Person = {
+    name: 'John Doe'
+  };
+  constructor() { }
+}
+```
+
+The `@Component` tag is used to mark a component. It provides the metadata that determines how the component is processed during runtime.
+
+* The use of `selector: app-person` is how Angular identifies this component in a template. The following code constitutes the template of `app.component.html`.
+
+    ```html
+    <h1>Person Data</h1>
+    <app-person></app-person>
+    <hr/>
+    ```
+
+    Angular identifies the `<app-person>` tag, instantiates the `PersonComponent` and renders the view defined by the template `./person.component.html`.
+
+* The `person.name` used in the template `./person.component.html`, refers to the name property defined in the above component.
+
+Now let us look at what **Dependency Injection**(DI) means. Dependencies are services that a class needs in order to perform it's function. We can visualize this with the help of an example.
```

Comment:
"perform it's function" --> "perform its functions"

Do you think you should add a link to the word "Dependency Injection" in case people want to read more info about it?

Reply:
Prof and I think its better not to so that the flow is maintained.